### ISSUE TITLE
Comments on the Accessibility Conformance Testing (ACT) Rules Format 1.0 Working Draft (6 April 2017 version)

### ISSUE THREAD

**Issue by RaphDeRooij** [w3c/wcag-act#82]
Dear editors of the ACT Rules Format,

First of all, I am positively surprised by the fact that W3C is developing a Rules Format specification, because in my opinion it can be very helpful when merging web accessibility test results from multiple resources. Merging may very well become common practice when the European web accessibility directive<sup>[1]</sup> is transposed by the member states, since using different testing methods is already common practice and the test results are an important component of the mandatory accessibility statements.

Web accessibility test results can be the output of different processes:
- fully automated test tools,
- semi-automated test procedures (automated finding of ‘suspects’, combined with evaluation by humans),
- manual evaluation by developers or editors, and
- manual evaluation by independent accessibility service providers.

Aggregation of test results may be necessary to assemble a complete set of test results, in accordance with the requirements of a referenced evaluation methodology<sup>[2]</sup>.

<h2>Comment 1: scope of the specification</h2>

According to the abstract, the working draft is developed for automated test tools only. My main question at this point is about the scope of the specification: is it also ‑ or can it easily be made ‑ suitable for the other three scenarios that are described in above paragraph? 
The ‘business case’ behind this question is that a specification such as ACT Rules Format is usable to ensure the reliability of assembling sets of partial test results.

<h2>Comment 2: outcome ‘Cannot tell’</h2>

Another question is about the ‘Cannot tell’ outcome. There may be various reasons for this outcome:
- the test may be not applicable
- it cannot be evaluated with a fully automated test tool,
- it requires human evaluation (example: ALT text is present, but does it match the image?)
- it can only be evaluated reliably by an expert
- ...

My question is whether it is possible to differentiate the ‘Cannot tell’ outcome in the specification. For example:
- Cannot tell – test may be not applicable
- Cannot tell – unfit for fully automated evaluation
- Cannot tell – human decision required
- Cannot tell – expert decision required
- ...

Many thanks to the editors for their work on the ACT Rules Format. My comments are intended to maximize the impact of this specification; in my opinion the ACT Rules Format has the potential to become an indispensable instrument in managing the compliance of web accessibility.

With kind regards,

Raph de Rooij
the Netherlands

-------
[1]: Available at (http://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX:32016L2102)
[2]: for example W3C’s Website Accessibility Conformance Evaluation Methodology (WCAG-EM) 1.0 (https://www.w3.org/TR/WCAG-EM/)

PS:
I found a typo in the text under ‘6.2. Outcome’, at the third bullet: ‘rails’ must be ‘fails’.

**Comment by dylanb** [comment]
> My question is whether it is possible to differentiate the ‘Cannot tell’ outcome in the specification. For > example:
>
> Cannot tell – test may be not applicable
> Cannot tell – unfit for fully automated evaluation
> Cannot tell – human decision required
> Cannot tell – expert decision required

A couple of comments:
1. Almost every automated test requires a matching human evaluation, so just because something passes the automated test, does not mean it is accessible. It only means that it passes the portion of the accessibility evaluation that can be performed fully automatically. E.g. alt text - a computer can test that the alt attribute exists, it cannot tell with accuracy whether it is the appropriate description for the image (note: even with improving image recognition, this will still hold for a long time). Given that this is true, the designation of "needs human review" is useful only in a rule that is explicitly designed to identify items that need to be reviewed by a human. So therefore, while I agree that the designations should be expanded, they should also be explicitly separated out into different rules. I.e. a rule cannot conditionally return "unfit for automation" or "needs human decision", it can only return "pass" or "needs human decision".
1. I am not sure what the difference is between needs human review and needs expert review. These seem like arbitrary points on a scale that is dependent on the human's skills. I think we will spend too much time arguing about what belongs to what skill set and it will remain arbitrary and therefore this distinction is not useful.
1. I am struggling to think of an example for "this test may not be applicable" or how this would be useful.

**Comment by WilcoFiers** [comment]
We spent a full call on Acc support, and I think we've made some progress. Just dropping some notes for myself to work off of:

- Allow rules to be "grouped" in such a way that if any one rule passes, the group passes
- Groups allow for more atomic rules, this is to avoid the mega-rules
- For any test, you can 'disable' rules in a group for reasons of accessibility support, which will result in that not being a valid solution within that particular test.

Potential downside:

- It adds complexity to users of rules. You can still treat rules as a black box, but groups is something you have to understand if you're using ACT Rules
- There is limited opportunity for finetuning rules to particular accessibility support issues
